src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Inlined/NullableDecimalAverageAggregationOperator.cs

```diff
@@ -118,8 +118,11 @@ protected override bool MoveNextCore(ref Pair<decimal, long> currentElement)
 
                     if (current.HasValue)
                     {
-                        sum += current.GetValueOrDefault();
-                        count++;
+                        checked
+                        {
+                            sum += current.GetValueOrDefault();
+                            count++;
+                        }
```

Comment:
Heh.  Unlikely, actually; [you can't use that many elements, remember](http://blogs.msdn.com/b/pfxteam/archive/2012/11/16/plinq-and-int32-maxvalue.aspx)?  (In theory it means we could remove the checking altogether, even from the non-nullable versions, but that makes me nervous.) 
I ran into this with `LongCount`, too.  Of course, having it wrap around would be a pretty bad bug (silent incorrect values)....